incubator/jaeger/templates/spark-cronjob.yaml

```diff
@@ -29,40 +29,42 @@ spec:
-              value: {{ .Values.storage.type }}
-            {{- if eq .Values.storage.type "cassandra" }}
-            - name: CASSANDRA_CONTACT_POINTS
-              value: {{ template "cassandra.contact_points" . }}
-            - name: CASSANDRA_KEYSPACE
-              value: {{ printf "%s_%s" "jaeger_v1" .Values.cassandra.config.dc_name | quote }}
-              {{- end }}
-            {{- if eq .Values.storage.type "elasticsearch" }}
-            - name: ES_NODES
-              value: {{ template "elasticsearch.client.url" . }}
-            - name: ES_NODES_WAN_ONLY
-              value: {{ .Values.storage.elasticsearch.nodesWanOnly | quote }}
-            - name: ES_PASSWORD
-              valueFrom:
-                secretKeyRef:
-                  name: {{ if .Values.storage.elasticsearch.existingSecret }}{{ .Values.storage.elasticsearch.existingSecret }}{{- else }}{{ include "jaeger.fullname" . }}-elasticsearch{{- end }}
-                  key: password
-            - name: ES_USERNAME
-              value: {{ .Values.storage.elasticsearch.user }}
-            {{- end }}
-            resources:
-{{ toYaml .Values.spark.resources | indent 14 }}
+            - name: {{ include "jaeger.fullname" . }}-spark
+              securityContext:
+                  {{- toYaml .Values.spark.securityContext | nindent 16 }}
+              image: {{ .Values.spark.image }}:{{ .Values.spark.tag }}
+              imagePullPolicy: {{ .Values.spark.pullPolicy }}
+              env:
+                - name: STORAGE
+                  value: {{ .Values.storage.type }}
+                {{- if eq .Values.storage.type "cassandra" }}
+                - name: CASSANDRA_CONTACT_POINTS
+                  value: {{ template "cassandra.contact_points" . }}
+                - name: CASSANDRA_KEYSPACE
+                  value: {{ printf "%s_%s" "jaeger_v1" .Values.cassandra.config.dc_name | quote }}
+                  {{- end }}
+                {{- if eq .Values.storage.type "elasticsearch" }}
+                - name: ES_NODES
+                  value: {{ template "elasticsearch.client.url" . }}
+                - name: ES_NODES_WAN_ONLY
+                  value: {{ .Values.storage.elasticsearch.nodesWanOnly | quote }}
+                - name: ES_PASSWORD
+                  valueFrom:
+                    secretKeyRef:
+                      name: {{ if .Values.storage.elasticsearch.existingSecret }}{{ .Values.storage.elasticsearch.existingSecret }}{{- else }}{{ include "jaeger.fullname" . }}-elasticsearch{{- end }}
+                      key: password
+                - name: ES_USERNAME
+                  value: {{ .Values.storage.elasticsearch.user }}
+                {{- end }}
+              resources:
+                {{ toYaml .Values.spark.resources | indent 16 }}
+          nodeSelector:
+            {{ toYaml .Values.spark.nodeSelector | indent 12 }}
+          tolerations:
+            {{ toYaml .Values.spark.tolerations | indent 12 }}
```

Comment:
Is node selector and tolerations needed for the pod security context? If not, please, send as a separate PR.

Reply:
I will wait to see if you let this one slide... 1 logical change per PR is definitely the way to go and I will strive for this moving forward. Just seeing that you took the time to go over the changes, maybe we could let this one slide :)

Reply:
I really appreciate what you've done in this PR and I would be happy to merge this in. But indeed, I'd prefer 3 separate PRs. I promise to review them quickly.